pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -351,6 +389,18 @@ func (iqr *IQR) Append(other *IQR) error {
 		return err
 	}
 
+	if iqr.mode == withRRCs && other.mode == withoutRRCs {
```

Comment:
You're handling only one of two cases. You also need to handle when iqr.mode is withoutRRCs and other.mode is withRRCs